akka-http-core/src/main/resources/reference.conf

```diff
@@ -698,6 +698,14 @@ akka.http {
     # `ignore` : just ignore the illegal characters in response header value
     illegal-response-header-value-processing-mode = error
 
+    # Configures the processing mode when encountering conflicting `Content-Type` headers in response.
+    #
+    # Supported mode:
+    # `error` : default mode, throw an ParsingException and terminate the processing
+    # `first` : use the value of the first `Content-Type` header
+    # `last`  : use the value of the last `Content-Type` header
+    conflicting-response-content-type-header-processing-mode = error
+
```

Comment:
Using `first` and `last` sounds a bit like using `random`. It is likely that when the server sends too many content-type headers, the order is the same but it wouldn't be surprising if the order of headers sent from the server changed in every call.

Maybe there should a list of precedences? A list with like: `["text/plain", "application/json"]` so `text/plain` takes precedence. It would be a bit more expensive but this expensive check to determine what header has higher precedence would only be invoked when the server sends an invalid response. I haven't put too much thought on this alternative, though.

Reply:
You make a fair point about `first`/`last` being just as good as `random`. I like the idea of a precedence list (perhaps as another option) but it gets a bit tricky, such as how to handle response with Content-Types not in the precedence list, etc.

So how about this: in situations with conflicting Content-Types, we set the entity to a default Content-Type (specified in the config) e.g. `application/octet-stream`, place all the parsed Content-Type headers from the response in the header list, and leave it up to the user to sort it all out on their own?